### ISSUE TITLE
Not using swap

### ISSUE THREAD

**Issue by abcdUwU** [janithcooray/lin_os_swap_mod#1]
Samsung A51 device with rom lineage os uses a very small amount of swap, in the amount of several megabytes, while there is 10 gigs (2 gigs system + 8 gigs swap magisk) of memory!

**Comment by janithcooray** [comment]
@abcdUwU Sorry about the late reply, but the swap takes a few hours to kick in. keep it for about 3 hours after opening a few apps and see if the issue persists. also mention which version of lineage is installed

Thank you for using this!

**Comment by abcdUwU** [comment]
Thanks for the answer, but I don't use lineage os anymore, but I think it was version 20, made by royna, which was specially made for the a51 phone. 
I tested up to 5 hours, but it did not use swap

**Comment by janithcooray** [comment]
i use the same rom i think, made by royna too. I have got another issue similar to this, ill be checking on this furthermore

**Comment by janithcooray** [comment]
@citizenserious 
I have released the v1.2 update. i have added a feature called `Set Swap Priority above Zram?` choose `yes` during installation on magisk. ill be prioritising this issue. 

Please be kind enough to check out whether the update has resolved this issue for you.

Thank you!

**Comment by citizenserious** [comment]
@janithcooray wow thanks for looking into this that fast. I am actively testing and will report back soon.

**Comment by citizenserious** [comment]
![Screenshot_20230824-044957_App Manager](https://github.com/janithcooray/lin_os_swap_mod/assets/62773866/2dd002f7-58cf-4fb1-9535-08b8f013fd78)
It took me while to get my system there but it seems to work. 
Thanks a lot <3

**Comment by janithcooray** [comment]
oh cool. would you be able to give me a list of apps (or sort of apps) that you use and how frequently so I can try to improve this module? I notice some apps try to stay in memory forcefully. for example - the Google search app.

ZRAM priority is currently set to about 80%, wonder if it will be better to set it higher.

I'll be keeping this issue open for a while.

@citizenserious Thanks for the update anyway!

**Comment by citizenserious** [comment]
I had to open many apps at once get the 8GB to be used.
Apps I frequently use are:
Neo Launcher
Neo Feed
Neo Store
Neo Backup
NewPipe Sponsorblock
MagicEarth
Signal
Eternity
Infinity
Fedilab
Iceraven
Spotify
AntennaPod

I think that's not what a regular users would use frequently😄
But those willing to use your module probably aren't regular user anyway. Hope you could work with this information also.

**Comment by janithcooray** [comment]
> I had to open many apps at once to get the 8GB to be used.
> 
> Apps I frequently use are:
> 
> Neo Launcher Neo Feed Neo Store Neo Backup NewPipe Sponsorblock MagicEarth Signal Eternity Infinity Fedilab Iceraven Spotify AntennaPod
> 
> I think that's not what a regular users would use frequently😄 But those willing to use your module probably aren't regular user anyway. Hope you could work with this information also.
> 
> Something popped into my mind while writing the list: Is it a security risk to publish which apps are used publicly? Could it be a potential target when someone knows your IP? Some could search for exploids couldn't they? Just being curious, maybe you know.

Thanks a lot. ill try to see how this apps behave and add some patches to force them to go into swap

---
As a DevOps engineer , and a bit of a white hat hacker myself ,here is my experience on what you have asked.

## Q - Is it a security risk to publish which apps are used publicly?

### Answer 1 - in case you are talking related to a magisk module -
when a module is installed, we can use the module itself to find out what have you installed. if the developer misuses this, they can do some shady things. magisk/SU is pretty powerful and can inject some sort of spyware. these days 
android has the apps data/ directory encrypted, so its a little bit hard now. but having SU this is not that difficult to 
overcome. if your apps use unencrypted data on SQLite like whatsapp, they can easily add a method to upload the data 
from your device and read your Whatsapp texts. 
  

### Answer 2 - if not,


randomly knowing it doesn’t really cause a risk unless the specific app you use has a know vulnerability

---
## Q - Could it be a potential target when someone knows your IP
Answer -

knowing your IP alone isn’t a big threat. this becomes an issue if you have any sensitive ports open. for example ssh-22 
without authentication.  or again your app as a know vulnerability. most websites you visit record you IP.

## Q - Some could search for exploids couldn't they?

  people can look up for exploitation methods without even knowing what you use.
  this is problematic if you have installed a malicious module / su enabled app in your device. we can’t really exploit with the name of the apps alone. unless the app you use has terrible security
  

## Proof this module is not malicious 

you can check inside the module file. its a pretty small read 🙂


If you want to learn more, you can look at things like `metasploit` to learn more about this. metasploit is a hacking tool that has few methods for android as well - even without root.

**Comment by janithcooray** [comment]
Patched on new V1.3

**Comment by astroanax** [comment]
> ![Screenshot_20230824-044957_App Manager](https://private-user-images.githubusercontent.com/62773866/263121694-2dd002f7-58cf-4fb1-9535-08b8f013fd78.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQ4Mzk3MjMsIm5iZiI6MTcyNDgzOTQyMywicGF0aCI6Ii82Mjc3Mzg2Ni8yNjMxMjE2OTQtMmRkMDAyZjctNThjZi00ZmIxLTk1MzUtMDhiOGYwMTNmZDc4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODI4VDEwMDM0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTI4MTc1MzJiNzE4ODQwZmQ4ZmQwYzU2OThjNDRlMDliZDRkYmNhZDdmMjVmNWI5ODFhMTk3YzBiMDllNzljZmMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0Ylue1kkCS7kbEsH89qTu1Om02WhCe1F_yo6f5pbLvw) It took me a while to get my system there but it seems to work. Thanks a lot <3

@citizenserious How do you get this screen of memory and swap usage? I know its visible from the command line via free/htop but I don't know how to view swap usage from the android GUI

**Comment by citizenserious** [comment]
@astroanax it is not an android GUI, I used [AppManager](https://github.com/MuntashirAkon/AppManager) click the burger menu and select "Running Apps"

**Comment by astroanax** [comment]
btw @janithcooray I seem to be having the exact same issue, latest version. I am on crdroid, android 14. When installing. I chose 8GB, swap higher than zram, and it displayed set swappiness to 99 but after reboot it was 20. (I have 4GB of RAM)

When I check ram usage after a while, it was around half a gig, but I manually changed swapinness to 99, and its a bit better now, swap usage is at 1G. Still not quite at ts peak.
